### PR TITLE
Pb 6779 knobs integration

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

https://jira.elasticpath.com/browse/PB-6779.

1. It adds the knobs plugin into the storybook project as a dependency and is brought in via the storybook configs.

2. A file `storybookUtils.tsx` is added to handle function callbacks. (could think about contributing to knobs repo?) Seems like missing functionality to the plugin.

3. All components and all stories will have knobs integrated

Linting:

- [x] No linting errors

Component Updates:

- [ ] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [x] Manual tests

1.) Test that knobs is available and functional for each component.

Documentation:

- [x] Requires documentation updates

### CODE REVIEW

File: storybook/.storybook/config.js [hunk 1]
```diff
@@ -5,12 +5,10 @@ import intl from 'react-intl-universal';
 import { addParameters } from '@storybook/react';
 import { themes } from '@storybook/theming';
 import elasticpath from './elasticpath';
+import { addDecorator } from '@storybook/react';
+import { withKnobs } from '@storybook/addon-knobs';
 
-// addParameters({
```
Comment (shaunmaharaj) [inline storybook/.storybook/config.js]: thanks for cleaning this

**Comment by shaunmaharaj** [comment]
@aChanEP This LGTM so far! Thanks for implementing. We can proceed with the rest of the components.

File: storybook/utils/storybookUtils.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
```
Comment (shaunmaharaj) [inline storybook/utils/storybookUtils.tsx]: Please add our copyright statement to the top here

File: storybook/utils/storybookUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @export
+ * @param {string} funcText - A function that is a passed in as a string.
+ */
+export function textToFunc(funcText:string) {
+    let decodedText:string = decodeHtml(funcText);
+    let selfInvokingFuncStr:string = makeSelfInvokingFuncString(decodedText);
+    eval(selfInvokingFuncStr);
+}
+
```
Comment (shaunmaharaj) [inline storybook/utils/storybookUtils.tsx]: missing newline char

File: storybook/.storybook/addons.js [hunk 4]
```diff
@@ -2,3 +2,4 @@ import 'storybook-readme/register';
 import '@storybook/addon-storysource/register';
 import '@storybook/addon-actions/register';
 import '@storybook/addon-links/register';
+import '@storybook/addon-knobs/register';
```
Comment (shaunmaharaj) [inline storybook/.storybook/addons.js]: missing new line char

File: storybook/utils/storybookUtils.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    return '(' + funcText + ')();'
+}
+
+/**
+ * gets rid of any special html characters and returns the intended user string.
+ *
+ * @param {string} html
+ * @returns
+ */
+function decodeHtml(html:string) {
```
Comment (shaunmaharaj) [inline storybook/utils/storybookUtils.tsx]: This isn't done automatically as part of displaying the story?

File: storybook/utils/storybookUtils.tsx (hunk 2 above)
Comment (aChanEP) [reply storybook/utils/storybookUtils.tsx]: resolved

File: storybook/utils/storybookUtils.tsx (hunk 3 above)
Comment (aChanEP) [reply storybook/utils/storybookUtils.tsx]: resolved

File: storybook/.storybook/addons.js (hunk 4 above)
Comment (aChanEP) [reply storybook/.storybook/addons.js]: resolved

File: storybook/utils/storybookUtils.tsx (hunk 5 above)
Comment (aChanEP) [reply storybook/utils/storybookUtils.tsx]: not available in knobs plugin

**Pull request merged**